### PR TITLE
[exec] create exec.whitelist file when exec binding or transformation is installed

### PR DESCRIPTION
**Pull request by kaikreuzer**
fixes https://github.com/openhab/openhab-addons/issues/7108

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/ExecWhitelistWatchService.java [hunk 1]
```diff
@@ -63,11 +63,12 @@ protected boolean watchSubDirectories() {
     }
 
     @Override
-    protected void processWatchEvent(@Nullable WatchEvent<?> event,  WatchEvent.@Nullable Kind<?> kind, @Nullable Path path) {
+    protected void processWatchEvent(@Nullable WatchEvent<?> event, WatchEvent.@Nullable Kind<?> kind,
+            @Nullable Path path) {
         if (path.endsWith(COMMAND_WHITELIST_FILE)) {
```
Comment (cpmeister) [inline bundles/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/ExecWhitelistWatchService.java]: Isn't path nullable? Shouldn't you do a null check?

File: bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationWhitelistWatchService.java [hunk 2]
```diff
@@ -63,11 +63,12 @@ protected boolean watchSubDirectories() {
...
     @Override
-    protected void processWatchEvent(@Nullable WatchEvent<?> event,  WatchEvent.@Nullable Kind<?> kind, @Nullable Path path) {
+    protected void processWatchEvent(@Nullable WatchEvent<?> event, WatchEvent.@Nullable Kind<?> kind,
+            @Nullable Path path) {
         if (path.endsWith(COMMAND_WHITELIST_FILE)) {
             commandWhitelist.clear();
-            try  {
-                Files.lines(path).forEach(commandWhitelist::add);
+            try {
+                Files.lines(path).filter(line -> !line.startsWith("#")).forEach(commandWhitelist::add);
```
Comment (cpmeister) [inline bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationWhitelistWatchService.java]: I'd trim the line before checking for #

**Comment by kaikreuzer** [comment]
Thanks @cpmeister, I've addressed your comments.

**Comment by 9037568** [comment]
Should you update the README to discuss the new feature for ignoring lines in the whitelist?

**Comment by kaikreuzer** [comment]
No, I think that's not worth to mention explicitly.

**Comment by kaikreuzer** [comment]
Hm, this change in the feature.xml isn't picked up for https://ci.openhab.org/view/Integration%20Builds%20(2.5.x)/job/openHAB2.5.x-Addons/ws/features/openhab-addons/src/main/feature/feature.xml/*view*/

@J-N-K Do you have any idea why this does not work?

**Comment by J-N-K** [comment]
https://github.com/openhab/openhab-addons/blob/1f89d98e64c0853b2405f316e05baad746a90bcf/features/openhab-addons/pom.xml#L54

I think config needs to be added here.

**Comment by kaikreuzer** [comment]
Good catch, thanks. I have created https://github.com/openhab/openhab-addons/pull/7186.

**Pull request merged**